HandsOnProject/app/src/main/java/com/example/handsonproject/MainActivity.kt

```diff
@@ -1,11 +1,159 @@
+    private var operation : Operation = Operation.MAIN_STATUS
+
     override fun onCreate(savedInstanceState: Bundle?) {
+
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = inflate(LayoutInflater.from(this))
+        setContentView(binding.root)
+        with(binding){
+            stash.text=""
+        }
+    }
+
+
+    fun onTapB1(view: View){
+        with(binding) { input.setText(input.getText().toString() + "1") }
+    }
+    fun onTapB2(view: View){
+        with(binding) { input.setText(input.getText().toString() + "2") }
+    }
+    fun onTapB3(view: View){
+        with(binding) { input.setText(input.getText().toString() + "3") }
+    }
+    fun onTapB4(view: View){
+        with(binding) { input.setText(input.getText().toString() + "4") }
+    }
+    fun onTapB5(view: View){
+        with(binding) { input.setText(input.getText().toString() + "5") }
+    }
+    fun onTapB6(view: View){
+        with(binding) { input.setText(input.getText().toString() + "6") }
+    }
+    fun onTapB7(view: View){
+        with(binding) { input.setText(input.getText().toString() + "7") }
+    }
+    fun onTapB8(view: View){
+        with(binding) { input.setText(input.getText().toString() + "8") }
+    }
+    fun onTapB9(view: View){
+        with(binding) { input.setText(input.getText().toString() + "9") }
+    }
+    fun onTapB0(view: View){
+        with(binding) { input.setText(input.getText().toString() + "0") }
+    }
+    fun onTapBEqual(view: View){
+        with(binding){
+            if (input.text.toString() == "" && stash.text.toString() ==  "" ) return
+            when(operation){
+                Operation.PLUS -> {
+                    val res=input.text.toString().toDouble()+stash.text.toString().toDouble()
+                    input.setText(res.toString())
+                    stash.setText("")
+                }
+                Operation.MINUS -> {
+                    val res=stash.text.toString().toDouble()-input.text.toString().toDouble()
+                    input.setText(res.toString())
+                    stash.setText("")
+                }
+                Operation.MULTIPLY -> {
+                    val res=input.text.toString().toDouble()*stash.text.toString().toDouble()
+                    input.setText(res.toString())
+                    stash.setText("")
+                }
+                Operation.DIVIDE -> {
+                    val divisor = input.text.toString().toDouble()
+                    val res=stash.text.toString().toDouble()/divisor
+                    if(divisor!=0.0) input.setText(res.toString())
+                    else {
+                        AlertDialog
+                            .Builder(this@MainActivity)
+                            .setTitle("Err")
+                            .setMessage("Division by zero is not allowed!")
+                            .show()
+                    }
```

Comment:
@shovian Not a bug, but I think `res` could be moved inside the if-clause like this:

```
                    if(divisor!=0.0) {
                        val res=stash.text.toString().toDouble()/divisor
                        input.setText(res.toString())
                    } else {
                        AlertDialog
                            .Builder(this@MainActivity)
                            .setTitle("Err")
                            .setMessage("Division by zero is not allowed!")
                            .show()
                    }
```

Reply:
Was implemented